server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -130,19 +129,14 @@ public BalancedShardsAllocator(Settings settings) {
 
     @Inject
     public BalancedShardsAllocator(Settings settings, ClusterSettings clusterSettings, WriteLoadForecaster writeLoadForecaster) {
```

Comment:
There is a `settings` object in `clusterSettings` so it is no longer necessary to pass it into constructor.

Reply:
Makes sense. Why not remove this now-unused constructor argument?